NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -187,6 +207,105 @@ public GroupPermission getPermissionforGroup(Group group){
 		return permhandle.getGroupPermission(group);
 	}
 	
+	// == GROUP INTERACTIONS ============================================================== //
+	
+	public List<UUID> getMembers(String groupname) {
+		Group group = getGroup(groupname);
+		if (group == null) { 
+			return Lists.newArrayList();
+		}
+		return group.getAllMembers();
+	}
+	
+	public List<UUID> getMembers(String groupname, PlayerType rank) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return Lists.newArrayList();
+		}
+		return group.getAllMembers(rank);
+	}
+	
+	public PlayerType getMemberRank(String groupname, UUID player) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return null;
+		}
+		return group.getMember(player);
+	}
+	
+	public boolean isMember(String groupname, UUID player) {		
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return false;
+		}
+		return group.isMember(player);
+	}
+	
+	public boolean hasAccess(String groupname, UUID player, PermissionType perm) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return false;
+		}
+		
+		GroupPermission perms = getPermissionforGroup(group);
+		PlayerType rank = group.getMember(player);	
+		
+		return perms.isAccessible(rank, perm);
+	}
+	
+	/**
+	 * @param groupname - String name of the group
+	 * @return false if the group can be operated on, true otherwise
+	 */
+	public boolean isDisciplined(String groupname) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return true;
+		}
+		
+		return group.isDisciplined();
+	}
+	
+	public boolean areLinked(String supername, String subname) {
+		Group supergroup = getGroup(supername);
+		Group subgroup = getGroup(subname);
+		if (supergroup == null || subgroup == null) {
+			return false;
+		}
+		
+		return Group.areLinked(supergroup, subgroup);
+	}
+	
+	public boolean link(String supername, String subname) {
+		Group supergroup = getGroup(supername);
+		Group subgroup = getGroup(subname);
+		if (supergroup == null || subgroup == null) {
+			return false;
+		}
+		
+		boolean success = Group.link(supergroup, subgroup);
+		if (success) {
+			groupManagerDao.addSubGroup(supername, subname);
```

Comment:
No, I think this wont update other shards at all currently. To get other shards to recache you will have to send a Mercury message, for example the way it's done here https://github.com/tyrothalos/NameLayer/blob/master/NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java#L132


Other shards will then get the message in the form of an event, which has it's listener here: https://github.com/tyrothalos/NameLayer/blob/master/NameLayerMain/src/vg/civcraft/mc/namelayer/listeners/MercuryMessageListener.java#L32


Easiest would be to just add a link/unlink message and completly recache the groups affected whenever a shards gets a message about those.